src/clarity-icons/extend-clarity-icons.ts

```diff
@@ -0,0 +1,98 @@
+import { IconTemplate } from "./icon-template";
+import { IconAlias } from "./icon-alias";
+import { SVG_IMG_ICON_TEMPLATES } from "./svg-img-icon-templates";
+import { SVG_ICON_TEMPLATES } from "./svg-icon-templates";
+
+let EXTENDED_ICON_TEMPLATES: IconTemplate = {};
+
+export class UserIcons {
+
+    private validateName(name: string): boolean {
+
+        return name.length !== 0;
+
+    }
+
+    private validateTemplate(template: string): boolean {
+
+        return template.length !== 0;
+
+    }
+
+    getExtendedShapes(): IconTemplate {
+
+        return EXTENDED_ICON_TEMPLATES;
+    }
+
+
+    private setIconTemplate(shapeName: string, shapeTemplate: string): void {
+
+        if (this.validateName(shapeName) && this.validateTemplate(shapeTemplate)) {
+
+            EXTENDED_ICON_TEMPLATES[ shapeName ] = shapeTemplate;
+        }
+
+    }
+
+    addIcons(icons: IconTemplate): void {
```

Comment:
This is how it works currently. You can either add a single icon or add multiple icons in an object. 

For example:
```
ClarityIcons.addIcons({facebook: `<svg>...</svg>`});
ClarityIcons.addIcons({linkedin: `<svg>...</svg>`});
```
or 
```
ClarityIcons.addIcons({
facebook: `<svg>...</svg>`, 
linkedin: `<svg>...</svg>`,
...
...
...
});
```